core/deployment/src/main/java/io/quarkus/runner/RuntimeClassLoader.java

```diff
@@ -76,7 +82,29 @@
     public RuntimeClassLoader(ClassLoader parent, List<Path> applicationClassesDirectories, Path frameworkClassesDirectory,
             Path transformerCache) {
         super(parent);
-        this.applicationClassDirectories = applicationClassesDirectories;
+        List<Path> nonEmptyAppClasses = new ArrayList<>();
+        try {
+            Map<String, Path> applicationClasses = new HashMap<>();
+            for (Path i : applicationClassesDirectories) {
```

Comment:
Just to be sure, this one contains all the directories, even the nested ones? Because, maybe I missed something, but I don't see any recursivity in what you did?

Reply:
I am going to remove this, I added this before the map but now we have the map it is not really nessesary.